src/nxt_unit_typetraits.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022-2024, Alejandro Colomar <alx@kernel.org>
+ */
+
+#ifndef _NXT_UNIT_TYPETRAITS_H_INCLUDED_
+#define _NXT_UNIT_TYPETRAITS_H_INCLUDED_
+
+
+#include <limits.h>
+
+
+#define nxt_widthof(T)      (sizeof(T) * CHAR_BIT)
+
+
+#define nxt_is_unsigned(x)  (((typeof(x)) -1) > 1)
```

Comment:
`(typeof(x))` is a cast.

Thisnk of:

```c
((cast) -1 > 1)
```

which is equivalent to

```c
((cast) -1) > 0
```

Now substitue with two types:

```c
((int) -1 > 0)   // of course, this is false
```

and

```c
((unsigned int) -1 > 0)
```
which is
```c
(UINT_MAX > 0)  // of course, true
```